.editorconfig

```diff
@@ -181,28 +181,37 @@ visual_basic_style_unused_value_assignment_preference = unused_local_variable:wa
 ### Configuration for IDE code style by diagnostic IDs ###
 [*.{cs,vb}]
 
-# IDE0004: Remove unnecessary cast
-dotnet_diagnostic.IDE0004.severity = warning
-
-# IDE0005: Remove unnecessary usings/imports
-dotnet_diagnostic.IDE0005.severity = warning
-
-# IDE0051: Remove unused private members (no reads or writes)
-dotnet_diagnostic.IDE0051.severity = warning
-
-# IDE0052: Remove unread private members (writes but no reads)
-dotnet_diagnostic.IDE0052.severity = warning
-
-# IDE0055: Fix formatting
-dotnet_diagnostic.IDE0055.severity = warning
+dotnet_analyzer_diagnostic.category-Style.severity = warning
 
 # IDE0066: Convert switch statement to expression
 dotnet_diagnostic.IDE0066.severity = suggestion
 
 # IDE0073: File header
-dotnet_diagnostic.IDE0073.severity = warning
 file_header_template = Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+# IDE005_gen: Remove unnecessary usings in generated code
+dotnet_diagnostic.IDE0005_gen.severity = silent
```

Comment:
📝 This is unnecessary. The diagnostic is not configurable, so this setting is ignored.

Reply:
@sharwell These are being flagged on the build without this setting, and it does seem to be configurable:

```
1>C:\roslyn-analyzers\artifacts\obj\Microsoft.CodeAnalysis.NetAnalyzers\Debug\netstandard1.3\.NETStandard,Version=v1.3.AssemblyAttributes.cs(2,1,3,25): warning IDE0005_gen: Using directive is unnecessary.
1>C:\roslyn-analyzers\artifacts\obj\Microsoft.CodeAnalysis.NetAnalyzers\Debug\netstandard1.3\Microsoft.CodeAnalysis.NetAnalyzers.InternalsVisibleTo.cs(11,1,12,25): warning IDE0005_gen: Using directive is unnecessary.
1>C:\roslyn-analyzers\artifacts\obj\Microsoft.CodeAnalysis.NetAnalyzers\Debug\netstandard1.3\Microsoft.CodeAnalysis.NetAnalyzers.AssemblyInfo.cs(11,1,12,25): warning IDE0005_gen: Using directive is unnecessary.
1>Microsoft.CodeAnalysis.NetAnalyzers -> C:\roslyn-analyzers\artifacts\bin\Microsoft.CodeAnalysis.NetAnalyzers\Debug\netstandard1.3\Microsoft.CodeAnalysis.NetAnalyzers.dll
```

Reply:
The diagnostic descriptor is created here:

https://github.com/dotnet/roslyn/blob/2869fb77525e6128243e7607760c2dbe75864721/src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs#L73-L89

It is not possible for this diagnostic to produce a build warning if the analyzer driver is behaving correctly.

Reply:
https://github.com/dotnet/roslyn/pull/43306 - I will file a tracking issue to remove this entry once we move to a compiler with that fix.

Reply:
#3495